File: Day-11/ros2_ws/src/day11_python_package/day11_python_package/publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class PublisherNode(Node):

    def __init__(self):
        super().__init__('day11_publisher')

        self.publisher = self.create_publisher(
            String,
            'package_topic',
            10
        )

        self.counter = 0

        self.timer = self.create_timer(
            1.0,
            self.publish_message
        )

    def publish_message(self):
        msg = String()
        msg.data = f'Hello from Day 11 Package: {self.counter}'

        self.publisher.publish(msg)

        self.get_logger().info(
            f'Published: {msg.data}'
        )

        self.counter += 1


def main(args=None):

    rclpy.init(args=args)

    node = PublisherNode()

    rclpy.spin(node)

    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: Day-12/ros2_ws/src/day12_cpp_package/src/publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using namespace std::chrono_literals;

class PublisherNode : public rclcpp::Node
{
public:
  PublisherNode()
  : Node("day12_publisher"), counter_(0)
  {
    publisher_ = this->create_publisher<std_msgs::msg::String>(
      "cpp_topic", 10);

    timer_ = this->create_wall_timer(
      1s,
      std::bind(&PublisherNode::publish_message, this));
  }

private:
  void publish_message()
  {
    auto message = std_msgs::msg::String();

    message.data =
      "Hello from Day 12 C++ Package: " + std::to_string(counter_);

    publisher_->publish(message);

    RCLCPP_INFO(
      this->get_logger(),
      "Published: %s",
      message.data.c_str());

    counter_++;
  }

  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
  int counter_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);

  auto node = std::make_shared<PublisherNode>();

  rclcpp::spin(node);

  rclcpp::shutdown();

  return 0;
}
